code/game/machinery/antigrav.dm

```diff
@@ -0,0 +1,124 @@
+/obj/machinery/antigrav
+	name = "antigrav generetor"
+	desc = "It temporarily disables gravity around."
+	icon_state = "GraviMobile"
+	density = 1
+	anchored = 0
+	use_power = 1
+	idle_power_usage = 10
+	active_power_usage = 10000
+	circuit = /obj/item/weapon/circuitboard/autolathe
+
+	var/on = FALSE
+	var/image/ball = null
+	var/area/area = null
+
+/obj/machinery/antigrav/proc/start()
+	area = get_area(src)
+	if(area.gravity_blocker && area.gravity_blocker != src)
+		src.visible_message(SPAN_NOTICE("\The [src] failed to start."))
+		return
+
+	area.gravity_blocker = src
+	area.has_gravity = FALSE
+
+	if(!on)
+		start_anim()
+
+	on = TRUE
+	use_power = 2
+	update_icon()
+
+/obj/machinery/antigrav/proc/stop()
+	if(area)
+		area.gravity_blocker = null
+		area.gravitychange(area.cached_gravity, area)
+		area = null
+
+	if(on)
+		stop_anim()
+
+	on = FALSE
+	use_power = 1
+	update_icon()
+
+/obj/machinery/antigrav/Process()
```

Comment:
There's no parent call here. IIRC the machinery/Process is where power usage is handled

Reply:
then why are there even checks for power there?

if power isnt used from process, theres no purpose in those checks. that would just make some things process forever if they use power despite not needing to